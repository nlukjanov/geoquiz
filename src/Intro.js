import React from 'react';
import { Link } from 'react-router-dom';


const Intro = () => {
  return (
    <div>
      <h1>Geo Quiz!</h1>
      <Link to='/game-select'>Start New Game</Link>
    </div>
  );
};

export default Intro;
