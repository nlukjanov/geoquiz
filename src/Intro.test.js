import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Intro from './Intro';

describe('<Intro />', () => {
  it('should render a component', () => {
    const { container, getByTestId } = render(
      <BrowserRouter>
        <Intro />
      </BrowserRouter>
    );
    expect(getByTestId('intro')).toBeTruthy;
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a h1 title', () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <Intro />
      </BrowserRouter>
    );
    // const title = getByTestId('h1 title');
    expect(getByTestId('h1 title')).toBeTruthy;
    expect(getByTestId('h1 title').textContent).toEqual('Geo Quiz!');
  });

  it('should have a button to start new game', () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <Intro />
      </BrowserRouter>
    );
    expect(getByTestId('link to new game').closest('a')).toHaveAttribute('href', '/game-select');
    expect(getByTestId('link to new game')).toBeTruthy;
    expect(getByTestId('link to new game').textContent).toEqual('Start New Game');
  });
});
