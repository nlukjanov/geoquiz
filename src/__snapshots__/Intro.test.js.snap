// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Intro /> should render a component 1`] = `
<div
  data-testid="intro"
>
  <h1
    class="title"
    data-testid="h1 title"
  >
    Geo Quiz!
  </h1>
  <a
    data-testid="link to new game"
    href="/game-select"
  >
    <button
      class="start"
    >
      Start New Game
    </button>
  </a>
</div>
`;
